tests/test_prompt.py

```diff
@@ -0,0 +1,285 @@
+"""Tests the xonsh Prompt Formatter."""
+from __future__ import unicode_literals, print_function
+
+import os
+import sys
+import copy
+import time
+import socket
+from datetime import datetime, timedelta
+from unittest import mock
+
+sys.path.insert(0, os.path.abspath('..'))  # FIXME
+
+import nose
+from nose.tools import eq_, ok_, assert_in, assert_is_instance
+from nose.plugins.skip import SkipTest
+
+from .tools import mock_xonsh_env
+from xonsh.tools import TERM_COLORS
+
+from xonsh.environ import (BasePrompt, DefaultPrompt,
+        DEFAULT_PROMPT, DEFAULT_TITLE, add_prompt_var)
+
+
+class Base_TestPrompt:
```

Comment:
Tests should be in a procedural style: http://xonsh.org/devguide.html#python-style-guide